src/index.js

```diff
@@ -0,0 +1,276 @@
+import "./pages/index.css";
+import Api from "./utils/Api.js";
+import {
+  enableValidation,
+  settings,
+  resetValidation,
+  disableButton,
+} from "./scripts/validation.js";
+
+// Profile elements
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const cardModalButton = document.querySelector(".profile__add-btn");
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+
+// Form elements
+const editModal = document.querySelector("#edit-modal");
+const editFormElement = editModal.querySelector(".modal__form");
+const editModalCloseBtn = editModal.querySelector(".modal__close-btn");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+const editSubmitButton = editModal.querySelector(".modal__submit-btn");
+
+const deleteConfirmModal = document.querySelector("#confirm-delete");
+console.log(deleteConfirmModal);
+const deleteConfirmCloseBtn =
+  deleteConfirmModal.querySelector(".modal__close-btn");
+
+const CancelConfirmCloseBtn = deleteConfirmModal.querySelector(
+  ".modal__cancel-button"
+);
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardModalCloseBtn = cardModal.querySelector(".modal__close-btn");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const cardSubmitButton = cardModal.querySelector(".modal__submit-btn");
+
+// Preview
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(".modal__close-btn");
+
+// Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+// Change avatar functionality
+const changeAvatarButton = document.querySelector(".profile__avatar-btn");
+const changeAvatarModal = document.querySelector("#edit-avatar-modal");
+const changeAvatarForm = changeAvatarModal.querySelector("#edit-avatar-form");
+const avatarSubmitButton =
+  changeAvatarModal.querySelector(".modal__submit-btn");
+const changeAvatarClose = changeAvatarModal.querySelector(".modal__close-btn");
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d44cd1da-3eb5-4df7-b9ca-8d742a5da7d3",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    const profileAvatar = document.querySelector(".profile__avatar");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine

Reply:
Another remark hasn't been corrected https://github.com/TheKunFo/se_project_spots/pull/1#discussion_r2083014924 

It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine